src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -400,4 +400,22 @@ class HttpGoogleServicesDAO(priceListUrl: String, defaultPriceList: GooglePriceL
     getScopedServiceAccountCredentials(firecloudAdminSACreds, authScopes)
       .refreshAccessToken()
       .getTokenValue
+
+  override def listBucket(bucketName: GcsBucketName,
+                          prefix: Option[String],
+                          recursive: Boolean = true
+  ): List[GcsObjectName] = {
+    // listObjectsWithPrefix handles paginating through results if there are more results than
+    // the `maxPageSize` setting.
+    val listAttempt = getStorageResource.use { storageService =>
+      storageService
+        .listObjectsWithPrefix(bucketName, prefix.getOrElse(""), maxPageSize = 2000, isRecursive = recursive)
```

Comment:
Could we make `maxPageSize` a parameter in the API to support a paging solution using `limit` and `offset`? This would help manage cases where there are a large number of files in a bucket.

Reply:
The `listObjectsWithPrefix` in workbench-libs handles pagination internally; it will automatically make multiple requests to ensure it lists the entire bucket. The `maxPageSize` here controls how large the individual requests are, but it will always list the entire bucket. Therefore, `maxPageSize` is just a performance knob. With that given, do you still think it's necessary to expose it as an API parameter?

Reply:
Thanks for clarifying! Since `maxPageSize` only affects performance and not the final result, keeping it within the implementation may be better than exposing it as an API parameter. This way, we can adjust it internally for efficiency without adding complexity to the API.